openscm/core/time.py

```diff
@@ -422,8 +422,11 @@ def __init__(  # pylint: disable=too-many-arguments
         self._interpolation_type = interpolation_type
         self._extrapolation_type = extrapolation_type
 
-        if self._source[0] > self._target[1]:  # TODO: consider extrapolation type
-            raise InsufficientDataError
+        if extrapolation_type == ExtrapolationType.NONE:
+            if self._source[0] > self._target[-1]:
```

Comment:
for ` ExtrapolationType.NONE` not a single missing time step can be extrapolated, so the check should stay as `self._source[0] > self._target[1]`, the other should be for `self._source[-1] < self._target[-2]`

Reply:
Should it not be self._source[0] > self._target[0] then? (I'm missing something but I can't see what...)